light/errors.go

```diff
@@ -58,6 +66,21 @@ func (e ErrVerificationFailed) Error() string {
 		e.From, e.To, e.Reason)
 }
 
+// ----------------------------- INTERNAL ERRORS ---------------------------------
+
+// ErrConflictingHeaders is thrown when two conflicting headers are discovered.
+type errConflictingHeaders struct {
+	Block   *types.LightBlock
+	Witness provider.Provider
```

Comment:
I think the reason for passing witness index instead of witness itself was me trying to avoid concurrency bugs. do we need Witness here?

Reply:
Yeah we need witness to know which provider should be the source of the `verifySkipping` function

Reply:
I guess we could extract out the witness from the index though